scriptworker/test/test_artifacts.py

```diff
@@ -377,3 +378,31 @@ def test_get_single_upstream_artifact_full_path(context):
 )))
 def test_get_optional_artifacts_per_task_id(upstream_artifacts, expected):
     assert get_optional_artifacts_per_task_id(upstream_artifacts) == expected
+
+
+# assert_is_parent {{{1
+@pytest.mark.parametrize("path, parent_path, raises", ((
+    "/foo/bar/baz", "/foo/bar", False
+), (
+    "/foo", "/foo/bar", True
+), (
+    "/foo/bar/..", "/foo/bar", True
+)))
+def test_assert_is_parent(path, parent_path, raises):
+    if raises:
+        with pytest.raises(ScriptWorkerTaskException):
+            swartifacts.assert_is_parent(path, parent_path)
+    else:
+        swartifacts.assert_is_parent(path, parent_path)
+
+
+def test_assert_is_parent_softlink(tmpdir):
```

Comment:
👍  for thinking of soft links. That makes me think, should we also test hard links? https://docs.python.org/2/library/os.html#os.link

Reply:
I don't think we can download a hardlink, while we might be able to download a softlink.